engines/sqlite.py

```diff
@@ -60,34 +63,45 @@ def insert_data_from_file(self, filename):
         Checks to see if a given file can be bulk inserted, and if so loads
         it in chunks and inserts those chunks into the database using
         executemany.
-
         """
         CHUNK_SIZE = 1000000
         self.get_cursor()
+        self.set_engine_encoding()
+        self.connection.text_factory = str
         ct = len([True for c in self.table.columns if c[1][0][:3] == "ct-"]) != 0
         if (([self.table.cleanup.function, self.table.header_rows] == [no_cleanup, 1])
-            and not self.table.fixed_width
-            and not ct
-            and (not hasattr(self.table, "do_not_bulk_insert") or not self.table.do_not_bulk_insert)
-            ):
+                and not self.table.fixed_width
+                and not ct
+                and not self.table.columns[0][1][0] == "pk-auto"
+                and (not hasattr(self.table, "do_not_bulk_insert") or not self.table.do_not_bulk_insert)
+                ):
             columns = self.table.get_insert_columns()
+            column_len = columns.split(self.table.delimiter)
             filename = os.path.abspath(filename)
             try:
                 bulk_insert_statement = self.get_bulk_insert_statement()
                 line_endings = set(['\n', '\r', '\r\n'])
-                with open(filename, 'r') as data_file:
+                with io.open(filename, 'rt', newline='', encoding='latin-1') as data_file:
                     data_chunk = data_file.readlines(CHUNK_SIZE)
-                    data_chunk = [line.rstrip('\r\n') for line in data_chunk if line not in line_endings]
+                    data_chunk = [line.rstrip('\r\n') for line in data_chunk if line.rstrip(
+                        '\r\n') not in line_endings]
                     del(data_chunk[:self.table.header_rows])
                     while data_chunk:
-                        data_chunk_split = [row.split(self.table.delimiter)
-                                            for row in data_chunk]
-                        self.cursor.executemany(bulk_insert_statement, data_chunk_split)
+                        data_chunk_split = []
+                        for row in data_chunk:
+                            row_values = row.split(self.table.delimiter)
+                            if len(row_values) != column_len:
+                                row_values = [None] * \
+                                    (len(column_len) - len(row_values))
```

Comment:
Is this the same behavior that we have for all the other engines? My naive assumption is that with data structured like this the missing fields are nulls, but the rest of the data is valid and should be included.